Provide more detail on how to get the Vagrant Cloud upload to work

jmcgeheeiv opened the pull request:
Thank you for the excellent example.

I had quite a bit of trouble getting this to work with Vagrant Cloud.  The main problem was no error message about my invalid Vagrant Cloud token, for which I even submitted hashicorp/packer#6866.  Wow.

There were a couple more minor struggles, and I documented them for you.

jmcgeheeiv committed:
Works.

jmcgeheeiv committed:
Template works and README accurate.

jmcgeheeiv committed:
Update README.md

azr reviewed (approved):
nice one.

azr commented on ubuntu1804.json:
```diff
@@ -1,5 +1,6 @@
 {
   "variables": {
+    "username": "",
```
when you set `"username": null,` then to quote [usage](https://www.packer.io/docs/templates/user-variables.html#usage) doc.
> If the default value is null, then the user variable will be required. This means that the user must specify a value for this variable or template validation will fail.

this is a nice feature to make sure your users set it by for example running `packer -var username=geerlingguy`

jmcgeheeiv committed:
Tell user to remove variables when not using Vagrant Cloud

jmcgeheeiv committed:
"With upload to Vagrant Cloud"

jmcgeheeiv committed:
Set user variable defaults to null

To cause an error when they are not set on the command line.

jmcgeheeiv commented:
As suggested by @azr:
* I changed the user variable defaults to `null` to make them required on the Packer command line
* Added instructions that the variables must be deleted when skipping Vagrant Cloud upload (so they are no longer required on the command line)

geerlingguy commented:
LGTM. I am going to merge this soon, but I will need to spend a little time updating my other base box configs as well (someday I'll unite them like Bento does...) before merging, just to keep things uniform across all my base box builds.

jmcgeheeiv committed:
Mention nfsd as a prerequisite

I accidentally also installed NIS.  In the absence of an NIS server,
this made my laptop behave very badly.  Thus the warning.

jmcgeheeiv committed:
Expanded instructions on running the Vagrant Boxes

jmcgeheeiv committed:
Corrected README heading depth

jmcgeheeiv committed:
We're not offering alternatives to VirtualBox

jmcgeheeiv committed:
Formatting

geerlingguy commented:
The latest commits make this more of a no-go for me—it's basically a complete rewrite of the documentation, and individually, I am willing to accept some of the changes to clarify bits and pieces of the build process. But as a whole, this makes the README really huge, and also makes this particular README file a lot different than the ones in all my other packer box builds.

Eventually I'm planning on merging them all into one repo ala Bento, but for now, it's hard enough trying to maintain consistent documentation :/

jmcgeheeiv commented:
Yes, I have battled documentation file skew quite a bit myself, so I understand.

Just merge whatever you like, whenever you like. I will not take offense.

geerlingguy commented:
This project is deprecated as things are being consolidated into https://github.com/geerlingguy/packer-boxes; see issue: https://github.com/geerlingguy/packer-boxes/issues/45
